libdnf5/base/goal.cpp

```diff
@@ -203,6 +205,9 @@ class Goal::Impl {
 
     void install_group_package(base::Transaction & transaction, libdnf5::comps::Package pkg);
     void remove_group_packages(const rpm::PackageSet & remove_candidates);
+
+    transaction::TransactionReplay transaction_replay;
+    GoalJobSettings transaction_replay_settings;
```

Comment:
What about to replace it by a vector with pair type inside. An empty vector should be cheaper then two default structures. What do you think?

Reply:
I think vector might be a little bit confusing if we only allow one replay per goal. I will try `std::unique_ptr`.

Reply:
Perfect.